Program.cs

```diff
@@ -404,6 +405,70 @@ static void Main(string[] args)
             var add = WorkingDayHelper.Instance.AddWorkingDays(DateTime.Today, 3);
             Debug.WriteLine(add); // output: 12/9/2022 12:00:00 AM
 
+            // 'Chain of Responsibility' pattern usage:
+            Customer customer = new Customer()
+            {
+                Name = "Carl Vosper Mortimer",
+                Age = 17
+            };
+            Request request = new Request() { Data = customer };
+            var ageHandler = new AgeHandler();
+            var nameHandler = new NameHandler();
+            ageHandler.SetTheNextHandler(nameHandler);
+            ageHandler.Process(request);
+            foreach (string message in request.ValidationMessages)
+            {
+                Debug.WriteLine(message);
+            }
+
+            // 'Abstract Factory' pattern usage:
+            HomeDevice homeDevice = null;
+            HomeDevicesFactory homeDevicesFactory = null;
+            homeDevicesFactory = HomeDevicesFactory.CreateHomeDevicesFactory("Kitchen");
+            Debug.WriteLine("Home Devices Factory type : " + homeDevicesFactory.GetType().Name);
+            homeDevice = homeDevicesFactory.GetDevice("Kettle");
+            Debug.WriteLine("Device Type : " + homeDevice.GetType().Name);
+            Debug.WriteLine(homeDevice.GetType().Name + " is " + homeDevice.feature());
+            homeDevicesFactory = HomeDevicesFactory.CreateHomeDevicesFactory("Cleaning");
+            Debug.WriteLine("Home Devices Factory type : " + homeDevicesFactory.GetType().Name);
+            homeDevice = homeDevicesFactory.GetDevice("Hoover");
+            Debug.WriteLine("Device Type : " + homeDevice.GetType().Name);
+            Debug.WriteLine(homeDevice.GetType().Name + " is " + homeDevice.feature());
+
+            // 'Factory Design' pattern usage:
+            string coffeeMachineMake = "Barista T Smart";
```

Comment:
the whole idea of the pattern was to "hide" the factory, while the communication to happen via the interface methods GetCoffeeMachineMake, GetNumberOfDrinksAvailable, GetMainFunction.

the double layered factory may look confusing to you at a first sight - the abstract class and concrete class with the override methods. but the point is that abstract layer always stay untouched, while if necessary changes will occur in the second layer, which secures the communication from the outside wont brake.